src/harness/testcases/WINNF_FT_S_DRG_testcase.py

```diff
@@ -269,8 +272,175 @@ def test_WINNF_FT_S_DRG_5(self):
+
+  @configurable_testcase(generate_DRG_6_default_config)
+  def test_WINNF_FT_S_DRG_6(self, config_filename):
+    """[Configurable] Array Deregistration."""
+
+    config = loadConfig(config_filename)
+    # Very light checking of the config file.
+    self.assertEqual(
+        len(config['registrationRequest']), len(config['grantRequest']))
+    self.assertTrue(
+        len(config['grantRequest']), len(config['heartbeatRequest']))
+    self.assertTrue(
+        len(config['heartbeatRequest']), len(config['deregistrationRequest']))
+    self.assertEqual(
+        len(config['deregistrationRequest']),
+        len(config['expectedResponseCodes']))
+
+    # Whitelist FCC IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectFccId({
+          'fccId': device['fccId'],
+          'fccMaxEirp': 47
+      })
+
+    # Whitelist user IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    # Step 2 & 3: Register devices and get grants
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(
+        config['registrationRequest'], config['grantRequest'],
+        config['conditionalRegistrationData'])
+
+    # Step 4: First Heartbeat Request
+    heartbeat_request = config['heartbeatRequest']
+    addCbsdIdsToRequests(cbsd_ids, heartbeat_request)
+    addGrantIdsToRequests(grant_ids, heartbeat_request)
+    request = {'heartbeatRequest': heartbeat_request}
+    responses = self._sas.Heartbeat(request)['heartbeatResponse']
+    self.assertEqual(len(responses), len(heartbeat_request))
+    # Check heartbeat response
+    for i, response in enumerate(responses):
+      self.assertEqual(response['response']['responseCode'], 0)
+    del request, responses
+
+    # Step 5: First deregistration request
+    degregister_request = config['deregistrationRequest']
+    addCbsdIdsToRequests(cbsd_ids, degregister_request)
+    request = {'deregistrationRequest': degregister_request}
+    responses_1 = self._sas.Deregistration(request)['deregistrationResponse']
+    # Check the deregistration response
+    self.assertEqual(len(responses_1), len(config['expectedResponseCodes']))
+    for i, response in enumerate(responses_1):
+      expected_response_codes = config['expectedResponseCodes'][i]
+      logging.debug('Looking at response number %d', i)
+      logging.debug('Expecting to see response code in set %s in response: %s',
+                    expected_response_codes, response)
+      self.assertIn(response['response']['responseCode'],
+                    expected_response_codes)
+      # "If the corresponding request contained a valid cbsdId, the
+      # response shall contain the same cbsdId."
+      if 'cbsdId' in degregister_request[i]:
+        if degregister_request[i]['cbsdId'] in cbsd_ids:
+          self.assertEqual(response['cbsdId'], degregister_request[i]['cbsdId'])
+        else:
+          self.assertFalse('cbsdId' in response)
+
```

Comment:
The spec mentions this   additional check "If responseCode == SUCCESS, verify that the response   contains a valid cbsdId.". I see that this check is redundant, but just   for the sake aligning with the test spec we should either add an additional   check here or have this statement removed from the test spec.

Reply:
Proposal to have this removed is underway. :)

Reply:
Thanks. we have verified the latest submission and have no more comments. I think this PR can be merged once the changes to test spec is approved.